python/cudf/cudf/core/column/categorical.py

```diff
@@ -56,9 +56,28 @@ def as_unordered(self, **kwargs):
 
     def add_categories(self, new_categories, **kwargs):
         new_categories = column.as_column(new_categories)
-        new_categories = self._column.categories.append(new_categories)
+        old_categories = self._column.categories
+
+        common_dtype = np.find_common_type(
+            [old_categories.dtype, new_categories.dtype], []
+        )
```

Comment:
Are we sure we want this behavior? I.E. if someone has a dictionary of `['a', 'b', 'c']` and calls `add_categories([1, 2, 3])` the result in Pandas would be `['a', 'b', 'c', 1, 2, 3]` but our result would be `['a', 'b', 'c', '1', '2', '3']` which is somewhat unexpected behavior.

Should we consider throwing on mixed dtype instead? That way if a user is okay with the behavior they can explicitly convert.

Reply:
Changed this behavior to raise an error for mixed types.